improver/cli/generate_timezone_mask_ancillary.py

```diff
@@ -0,0 +1,84 @@
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to run GenerateTimezoneMask ancillary generation."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(
+    cube: cli.inputcube, *, ignore_dst=True, time=None, groupings: cli.inputjson = None
+):
+    """Generate a timezone mask ancillary.
+
+    Create masks for regions of a geographic grid that are on different
+    timezones. The resulting masks can be used to isolate data for specific
+    timezones from a grid of the same shape as the cube used in this plugin
+    to generate the masks.
+
+    Args:
+        cube (iris.cube.Cube):
+            A cube with the desired grid. If no 'time' argument is provided
+            to the CLI, the time on this cube will be used for determining the
+            validity time of the calculated UTC offsets (this is only relevant
+            if daylights savings times are being included).
+        ignore_dst (bool):
+            If True, find and use the UTC offset to a grid point ignoring
+            daylights savings. Default is True.
+        time (str):
+            A datetime specified in the format YYYYMMDDTHHMMZ at which to
+            calculate the mask (UTC). If daylights savings are ignored this
+            will have no impact on the resulting masks.
+        groupings (dict):
+            A dictionary specifying how timezones should be grouped if so
+            desired. This dictionary takes the form::
+
+                {-9: [-12, -6], 0: [-5, 5], 6: [6, 12]}
+
+            The keys indicate the UTC offset that should provide the data for a
+            group. The numbers in the lists denote the inclusive limits of the
+            timezones for which that data should be used. This is of use if data
+            is not available at hourly intervals.
+
+    Returns:
+        iris.cube.Cube:
+            A cube containing timezone masks for each timezone found in the
+            grid within the input cube, grouped into larger areas if so desired.
```

Comment:
It might be worth adding something here that 1 means masked out and 0 means masked in, within the cube data.

Reply:
Done.